pkg/logs/auditor/auditor.go

```diff
@@ -34,28 +33,22 @@ type RegistryEntry struct {
 
 // An Auditor handles messages successfully submitted to the intake
 type Auditor struct {
-	inputChan     chan message.Message
-	registry      map[string]*RegistryEntry
-	registryMutex *sync.Mutex
-	registryPath  string
+	inputChan    chan message.Message
+	registry     map[string]*RegistryEntry
+	registryPath string
 
-	flushTicker   *time.Ticker
-	flushPeriod   time.Duration
-	cleanupTicker *time.Ticker
-	cleanupPeriod time.Duration
-	entryTTL      time.Duration
+	entryTTL time.Duration
+
+	isFlushed chan struct{}
 }
 
 // New returns an initialized Auditor
 func New(inputChan chan message.Message) *Auditor {
 	return &Auditor{
-		inputChan:     inputChan,
-		registryPath:  filepath.Join(config.LogsAgent.GetString("run_path"), "registry.json"),
-		registryMutex: &sync.Mutex{},
-
-		flushPeriod:   defaultFlushPeriod,
-		cleanupPeriod: defaultCleanupPeriod,
-		entryTTL:      defaultTTL,
+		inputChan:    inputChan,
+		registryPath: filepath.Join(config.LogsAgent.GetString("run_path"), "registry.json"),
+		entryTTL:     defaultTTL,
+		isFlushed:    make(chan struct{}),
```

Comment:
size 1?

Reply:
We could but it's not useful here, we add a capacity to channels when we don't want to block the go routine sending an event to the channel:
```
c := make(chan struct{}, 1)
c <- struct{}{} // not blocking
```
```
c := make(chan struct{})
c <- struct{}{} // blocking  until someone reads the channel
```
Here we know that it's not blocking because we already have another go routine waiting to receive the event.